frontend/style.css

```diff
@@ -16,132 +18,211 @@ html, body {
 /* Radiosity menu */
 #menu {
   position: absolute;
-  top: 15px;
-  left: 15px;
+  top: 1em;
+  left: 1em;
 
   border: solid 2px var(--fg);
 
-  font-size: larger;
-  font-family: 'Courier', Arial, monospace;
-  font-weight: bold;
+  font-family: 'Consolas', monospace;
+}
+
+#menu h2,
+#menu h3 {
+  font-size: 1em;
+}
+
+#menu h2 {
+  text-decoration: underline;
+  margin-bottom: .5em;
+  margin-top: .5em;
+}
+
+#menu h2:first-child {
+  margin-top: 0;
 }
 
 .menu-item {
   position: relative;
-  height: 40px;
-  width: 40px;
+  cursor: pointer;
+  border: 2px solid transparent;
+}
+
+.menu-item:hover,
+.menu-item.selected {
+  border-color: var(--bg);
 }
 
 /* Images */
-img {
-  margin-top: 7px;
-  margin-left: 7px;
+.menu-item img,
+.menu-item > i {
+  font-size: 24px;
+  display: block;
+  padding: 0.5rem;
+  color: var(--bg);
+  text-align: center;
 }
 
 /* Windows */
 .window {
+  position: absolute;
+  left: 150%; /* for compatibility if calc doesn't work */
+  left: calc(100% + 1em + 4px); /* 1em from the right of the menu, incl. borders */
+  top: -2px; /* to align with the border */
+
   border: solid 2px var(--fg);
+  background-color: var(--bga);
 
-  margin-left: 50px;
-  padding-left: 5px;
-  padding-top: 5px;
+  padding: 0.5em;
 
-  height: auto;
-  width: 300px;
+  min-width: 30em;
+  cursor: default; /* menu items are clickable, .windows aren't */
+}
 
-  background-color: var(--bg);
+.menu-item:first-child .window {
+  top: -4px; /* to align with both borders */
 }
 
-/* menu button colors */
-#menu > :nth-child(1) {
-  background-color: #05668d;
+.menu-item:not(.selected) .window {
+  display: none !important;
 }
 
-#menu > :nth-child(2) {
-  background-color: #028090;
+.option.selected {
+  font-weight: bold;
 }
 
-#menu > :nth-child(3) {
-  background-color: #00a896;
+.option.selected:after {
+  content: ' ◄';
+  display: inline;
 }
 
-#menu > :nth-child(4) {
-  background-color: #02c39a;
+/* menu button colors */
+#menu > :nth-child(1) { background-color: #75c9c8; }
```

Comment:
At first, I was really mixed about using child selector; for me, I thought it could be a bit confusing.
But in the end, I convinced myself that it's much more clean toassign colours this way if we have a lot of menu items.

Reply:
yes; it makes sense, I think, to have the colors in order of children, rather than assigned to specific menu items – when we reorder menu items, we are unlikely to want to change the colors